posthog/api/utils.py

```diff
@@ -31,7 +32,7 @@ class PaginationMode(Enum):
 
 def get_target_entity(filter: Union[Filter, StickinessFilter]) -> Entity:
     if not filter.target_entity_id:
-        raise ValueError("An entity id and the entity type must be provided to determine an entity")
+        raise ValidationError("An entity id and the entity type must be provided to determine an entity")
```

Comment:
Hmmm, should this knowledge be in the `Entity` constructor and this method catch a `TypeError` and convert it to `Validation`

Reply:
Maybe. I wanted to avoid making too big a change affecting the `Filter` code however as I wasn't sure how, e.g. validation works in cases where these values aren't actually required, given the `Filter` is used for multiple endpoint schemas.

I'll try adding it there though and fingers crossed the tests will pick up any issues.

Reply:
@pauldambra could I get this in as a little improvement as is then tackle the larger issue if and when I have time? I will do whatever you say.

Reply:
> I will do whatever you say.

🤔 I wonder how I can best take advantage of this :D

Yep, happy to mark my thinking aloud as bike shedding and you get this in to help the customer :)

Reply:
Great, I'll look into the Filter code as a whole separately.